backend/service/topology/cache.go

```diff
@@ -99,64 +100,110 @@ func (c *client) startTopologyCache(ctx context.Context) {
 	<-ctx.Done()
 }
 
+// Processes all topology cache update requests.
+// This supports insert batching which defaults to 1, and can be configured by settting BatchInsertSize.
+// Notably a flush will happen every 30 seconds, ensuring all items make it into cache even if the
+// configured BatchInsertSize is not met.
 func (c *client) processTopologyObjectChannel(ctx context.Context, objs <-chan *topologyv1.UpdateCacheRequest, service string) {
-	for obj := range objs {
-		c.scope.Tagged(map[string]string{
-			"service": service,
-		}).SubScope("cache").Gauge("object_channel.queue_depth").Update(float64(len(objs)))
-
-		switch obj.Action {
-		case topologyv1.UpdateCacheRequest_CREATE_OR_UPDATE:
-			if err := c.setCache(ctx, obj.Resource); err != nil {
-				c.log.Error("Error setting cache", zap.Error(err))
+	var batchInsert []*topologyv1.Resource
+
+	for {
+		select {
+		case obj, ok := <-objs:
+			if !ok {
+				return
+			}
+
+			c.scope.Tagged(map[string]string{
+				"service": service,
+			}).SubScope("cache").Gauge("object_channel.queue_depth").Update(float64(len(objs)))
+
+			switch obj.Action {
+			case topologyv1.UpdateCacheRequest_CREATE_OR_UPDATE:
+				batchInsert = append(batchInsert, obj.Resource)
+			case topologyv1.UpdateCacheRequest_DELETE:
+				if err := c.deleteCache(ctx, obj.Resource.Id, obj.Resource.Pb.TypeUrl); err != nil {
+					c.log.Error("Error deleting cache", zap.Error(err))
```

Comment:
i think there's an obscure bug here, not sure it's worth worrying about for now.

but let's say i get one insert and my batch size is 10. and then i get a delete every 15 seconds. the insert will never occur.

Reply:
Yeah i see what your saying, im going to leave a todo and circle back.